### PR TITLE
Add support for EC JWKs

### PR DESCRIPTION
**Pull request by jaspervdj**
This adds support for constructing JWKs from X509 EC certificates.

Feel free to tear this PR apart, and let me know what you need to get this
merged.  I'm happy to make changes to naming, structure and add tests.

### CODE REVIEW

**Comment by frasertweedale** [comment]
@jaspervdj it looks fine to me.  A test would be nice.  Hypocritical of me to ask because it seems I don't have any existing test for X.509-to-JWK behaviour.  But, since you offered... :)

File: src/Crypto/JOSE/Types.hs [hunk 1]
```diff
@@ -128,6 +129,14 @@ genSizedBase64IntegerOf :: Int -> Gen SizedBase64Integer
 genSizedBase64IntegerOf n =
   SizedBase64Integer n . bsToInteger <$> genByteStringOf n
 
+-- | Create a 'SizedBase64Integer'' from an 'Integer'.  This is slow since it
+-- does an encoding of the 'Integer' to check the size, so it should be used
+-- with care.
+makeSizedBase64Integer :: Integer -> SizedBase64Integer
+makeSizedBase64Integer x =
+  let bytes = review base64url (integerToBS x) in
```
Comment (frasertweedale) [inline src/Crypto/JOSE/Types.hs:137]: Hm, is this correct? - I think it is the number of _unencoded_ octets.

Comment (frasertweedale) [reply src/Crypto/JOSE/Types.hs:137]: Furthermore, should not need to encode to determine the length - the minimum required length is based on the logarithm of the integer value.

Comment (frasertweedale) [reply src/Crypto/JOSE/Types.hs:137]: See `instance Arbitrary SizedBase64Integer` from `Crypto.JOSE.Types`:
```haskell
arbitrary = do                                                          
  x <- arbitraryBigInteger                                              
  l <- Test.QuickCheck.elements [0,1,2]  -- number of leading zero-bytes
  pure $ SizedBase64Integer ((log2 x `div` 8) + 1 + l) x                
```

Comment (jaspervdj) [reply src/Crypto/JOSE/Types.hs:137]: You're right, sorry, I completely misread that part of the code.

**Comment by frasertweedale** [comment]
Thank you very much @jaspervdj.  I'll cut a new release in the next day or so.

**Comment by jaspervdj** [comment]
@frasertweedale Oh!  I actually still had a test coming up.  I'll open a separate PR.

**Comment by frasertweedale** [comment]
@jaspervdj v0.8.3 released: https://hackage.haskell.org/package/jose-0.8.3 

Many thanks for your contribution.

P.S. Thanks for hakyll :)

**Pull request merged**